SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -765,18 +805,7 @@ int64 USpatialActorChannel::ReplicateActor()
 				}
 				else
 				{
-					Sender->SendAuthorityIntentUpdate(*Actor, NewAuthVirtualWorkerId);
-
-					// If we're setting a different authority intent, preemptively changed to ROLE_SimulatedProxy
-					Actor->Role = ROLE_SimulatedProxy;
-					Actor->RemoteRole = ROLE_Authority;
-
-					Actor->OnAuthorityLost();
-
-					if (NetOwner != nullptr)
-					{
-						ForceReplicateOnActorHierarchy(NetDriver, NetOwner, Actor);
-					}
+					MarkActorHierarchyForMigration(NetDriver, NewAuthVirtualWorkerId, HierarchyRoot);
```

Comment:
If the Actor isn't part of a hierarchy (no owner or child Actors), we can avoid needing to delay at all, which seems preferable

Reply:
Issue is we don't know for sure, in case of teleport for example, maybe the owner will arrive later and locally resolve to null in the meantime.

Reply:
I don't think this is true? I think there's a difference between the owner and child references being unresolved or null?

Reply:
Yes, but that is not an information we can retrieve by merely checking the owner pointer. We would have to visit the properties involved in the hierarchy, and check if we recorded some UnrealObjectRef for them.